app/models/cherrypick/volume_by_nano_grams_per_micro_litre.rb

```diff
@@ -1,36 +1,68 @@
-module Cherrypick::VolumeByNanoGramsPerMicroLitre # rubocop:todo Style/Documentation
+# Included in {Well} to provide calculations for cherrypicking
+module Cherrypick::VolumeByNanoGramsPerMicroLitre
+  #
+  # Used in the cherrypicking process to calculate the relative volumes of source
+  # and buffer required to reach the target volume and concentration.
+  #
+  # - Ideally will aim for a target well containing volume_required at concentration_required
+  # - This will be made by combining material from the source well (picked_volume) and
+  #   buffer (buffer_volume)
+  # - Maximum picked_volume is limited by source_volume and volume_required
+  # - If source_volume < volume_required then buffer will be added to make up volume_required
+  #   even if it reduces the target concentration below concentration_required
+  #
+  # @param volume_required [Float] The volume to aim for in the target well (self)
+  # @param concentration_required [Float] The concentration to aim for in the target well (self)
+  # @param source_concentration [Float] The concentration (ng/ul) in the source well
+  # @param source_volume [Float] The volume (ul) in the source well (ie. the maximum pick)
+  # @param robot_minimum_picking_volume [Float] (ul) the minimum volume the robot can pick
+  #
+  # @return [Float] The total volume that will be picked from the source well
+  #
   def volume_to_cherrypick_by_nano_grams_per_micro_litre(volume_required, concentration_required, source_concentration, source_volume, robot_minimum_picking_volume = 0.0)
     robot_minimum_picking_volume ||= 0
 
     check_inputs_to_volume_to_cherrypick_by_nano_grams_per_micro_litre!(volume_required, concentration_required,
                                                                         source_concentration)
 
+    # @note Here we appear to set the concentration based on the required concentration, regardless of whether we hit it
+    # it or not. Checking if this behaviour is desired RT#719205
     well_attribute.concentration    = concentration_required
     well_attribute.requested_volume = volume_required
+    # Similarly we set current volume based on required. This is only untrue in rare edge cases though
+    # (When you have almost all your required volume from your source, then add more buffer than intended
+    #  due to minimum robot picks)
     well_attribute.current_volume   = volume_required
```

Comment:
This line is not part of the changes, but I dont see a match: volume_required  is what is intended to pick, but not necessarily will happen? Then current_volume is going to be that but there will be cases where the real current_volume will be: well_attribute.buffer_volume+ well_attribute.picked_volume. So picked_volume_buffer_volume is what really defines the current volume?